File: BST to GST/bstToGst.java
class Solution {
    public TreeNode bstToGst(TreeNode root) {
        reverse_in_order(root,0);
        return root ; 

        
    }
    private int reverse_in_order(TreeNode node , int acc){
        if(node == null){
            return acc;
        }
        acc=reverse_in_order(node.right,acc);
        acc+=node.val ;
        node.val = acc ; 
        acc=reverse_in_order(node.left,acc) ; 
        return acc ; 
    }
}

File: BST to GST/bstToGst.cpp
class Solution {
public:
    TreeNode* bstToGst(TreeNode* root) {
        reverseInOrder(root, 0);
        return root;
    }

private:
    int reverseInOrder(TreeNode* node, int acc) {
        if (node == nullptr) {
            return acc;
        }
        acc = reverseInOrder(node->right, acc);
        acc += node->val;
        node->val = acc;
        acc = reverseInOrder(node->left, acc);
        return acc;
    }
};
